config/configgrpc/configgrpc_test.go

```diff
@@ -796,7 +797,9 @@ func TestContextWithClient(t *testing.T) {
 	for _, tC := range testCases {
 		t.Run(tC.desc, func(t *testing.T) {
 			cl := client.FromContext(contextWithClient(tC.input, tC.doMetadata))
-			assert.Equal(t, tC.expected, cl)
+			assert.Equal(t, tC.expected.Addr, cl.Addr)
+			assert.Equal(t, tC.expected.Auth, cl.Auth)
+			assert.True(t, reflect.DeepEqual(tC.expected.Metadata, cl.Metadata))
```

Comment:
Does assert.Equal not do enough?

Reply:
Oh that was a holdover from the sync.Map attemp, good eye